### PR TITLE
Issue #84 - Detect provider and change provider name in user alerts

### PR DESCRIPTION
**Pull request by jordajm**
### Checklist:

- [x] Test your work and double check you didn't break anything

### Description:

This PR adds a util called `getCurrentProvider` and implements it in four files where the provider name `MetaMask` was hard-coded. Now, the provider name is determined dynamically so that users that are on providers other than MetaMask will see their provider's name in the alerts.

Also, this may be blocked by the pending updated way to reference `web3` in demo-dapp as noted here: https://github.com/OriginProtocol/demo-dapp/issues/126

### CODE REVIEW

File: src/components/web3-provider.js [hunk 1]
```diff
@@ -147,7 +150,7 @@ class Web3Provider extends Component {
     curr = curr && curr.toLowerCase()
 
     if (curr !== next) {
-      curr && Store.dispatch(showAlert('MetaMask account has changed.'))
+      curr && Store.dispatch(showAlert('{this.state.currentProvider} account has changed.'))
```
Comment (micahalcorn) [inline src/components/web3-provider.js]: Looks like this is supposed to be a template literal.

File: src/utils/getCurrentProvider.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export default function getCurrentProvider(web3) {
+    if (!web3 || !web3.version) {
```
Comment (micahalcorn) [inline src/utils/getCurrentProvider.js]: Please also convert this to two-space tabs instead of four-space. ✌️

**Comment by jordajm** [comment]
@micahalcorn good catches - thanks!

**Comment by jordajm** [comment]
Fixed and ready for another look @micahalcorn

**Comment by micahalcorn** [comment]
Thanks, @jordajm. Out of curiosity were you able to test this with any providers other than MetaMask? We could do it on mobile by deploying this branch to Heroku, but that seems like overkill.

**Comment by micahalcorn** [comment]
@jordajm I haven't forgotten about this. I was just trying to test on mobile and found other issues.

**Comment by jordajm** [comment]
@micahalcorn no worries - how are you doing mobile testing? I usually just use BrowserStack but I don't think I'd be able to test wallet scenarios there... do you have an easy way to deploy this branch to Heroku or are you running localhost on your phone?

**Comment by micahalcorn** [comment]
@jordajm currently Heroku, which seems to be the easiest way to test across multiple browsers. But I've yet to find a mobile wallet/browser that will result in the appropriate scenario to test one of the messages. I'm distracted with other (unrelated) issues for now. 😫

**Comment by jordajm** [comment]
@micahalcorn do you have this branch deployed to Heroku?

**Comment by micahalcorn** [comment]
@jordajm I honestly don't remember what is deployed. 😭 I'll let you know when I do.

**Comment by jordajm** [comment]
Ok let me know if I can help with testing - but without being able to deploy the branch to Heroku I don't think I'll be much help.

**Comment by micahalcorn** [comment]
This is great @jordajm 👍 I have tested it on all of the known providers except for Parity, which I can't make sense of. I suggest that we merge this and then worry about what recommendations to make based on which browser a person is using when there is no given provider (i.e., don't tell them to install MetaMask when on Safari).

  | Not Web3 Enabled | Not Signed In To Wallet | Unsupported Network
-- | -- | -- | --
MetaMask | N/A | ✅ | ✅
Trust | N/A | N/A | ✅
Toshi | N/A | N/A | ✅
Cipher | N/A | N/A | ✅
Mist | N/A | ✅ | ✅
Parity | ❔ | ❔ | ❔
Infura | N/A | N/A | ✅
Localhost | N/A | N/A | ✅

**Pull request merged**